### PR TITLE
CI: Try with sqlalchemy warnings

### PR DESCRIPTION
**Pull request by phofl**
- [ ] xref #50558 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mroeschke** [comment]
Thanks. Would the goal be to fully address https://github.com/pandas-dev/pandas/issues/50558?

**Comment by phofl** [comment]
Not sure if the con.begin() will fix everything. If yes, then this should cover the issue.

**Comment by phofl** [comment]
cc @mroeschke I am optimistic that this will do it (hopefully :))

Do we want to keep the warnings active on main as well till we can test against alchemy 2.0 or should I remove the env variable again?

**Comment by mroeschke** [comment]
I'm okay keeping the sqlalchemy warnings active on main (should hopefully aspire for sqlalchemy 2.0 compatibility so keeping these active will hopefully help)

**Comment by phofl** [comment]
CircleCI is green, this is a good sign :)

cc @MarcoGorelli

File: pandas/tests/io/test_sql.py [hunk 1]
```diff
@@ -879,11 +893,11 @@ def load_test_data_and_sql(self):
         create_and_load_iris_view(self.conn)
 
     def test_read_sql_view(self):
-        iris_frame = sql.read_sql_query("SELECT * FROM iris_view", self.conn)
+        iris_frame = sql.read_sql_query(self.text("SELECT * FROM iris_view"), self.conn)
```
Comment (lithomas1) [inline pandas/tests/io/test_sql.py:896]: Is this something that we actually need to fix in ``read_sql_query`` itself?

str is one of the accepted values for the ``sql`` parameter.

Comment (phofl) [reply pandas/tests/io/test_sql.py:896]: Can you open an issue about this? I would prefer pushing this to the user, because otherwise we would have to deal with SQLite without alchemy vs cases with alchemy

File: pandas/tests/io/test_sql.py [hunk 2]
```diff
@@ -2397,6 +2439,10 @@ def setup_driver(cls):
         # sqlite3 is built-in
         cls.driver = None
 
+    @pytest.mark.skip(reason="sorted on strings and integers")
+    def test_read_sql_parameter(self):
+        self._read_sql_iris_parameter()
```
Comment (MarcoGorelli) [inline pandas/tests/io/test_sql.py:2444]: why is this necessary?

**Review by MarcoGorelli — COMMENTED** [review]
just got a question about something, but the rest looks good, thanks so much for doing this!

did you manage to run these sqlalchemy tests locally BTW?

File: pandas/tests/io/test_sql.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/io/test_sql.py:2444]: There is some weird sort going on on the parameters to that query. They contain integers and strings which causes an error

**Comment by phofl** [comment]
The lite tests yes, I don’t have MySQL or Postgres set up locally…

**Comment by phofl** [comment]
Did you tackle the text thing as well?

**Comment by jbrockmendel** [comment]
i think ive got postgres running on my server.  would it help if i test this branch there?

**Comment by phofl** [comment]
I am leaning towards not fixing the auto commit problems now, since @cdcadman mentioned that he has a branch tackling these. The text problems are all fixed. 

But generally speaking, help is welcome here :)

**Comment by cdcadman** [comment]
@phofl , my branch only needed one modification due to 1.4.46 warnings, and I fixed it.  I don't mind if you use anything from that branch here: #48576 .
